subsys/dfu/boot/mcuboot_shell.c

```diff
@@ -91,6 +92,22 @@ static int cmd_mcuboot_erase(const struct shell *sh, size_t argc,
 
 	id = strtoul(argv[1], NULL, 0);
 
+#if FIXED_PARTITION_EXISTS(boot_partition) || DT_FIXED_PARTITION_EXISTS(DT_CHOSEN( \
+							zephyr_code_partition))
+	/* Check if this is the parent (MCUboot) or own slot and if so, deny the request */
+	if (0
+#if FIXED_PARTITION_EXISTS(boot_partition)
+	    || id == FIXED_PARTITION_ID(boot_partition)
+#endif
+#if DT_FIXED_PARTITION_EXISTS(DT_CHOSEN(zephyr_code_partition))
+	    || id == DT_FIXED_PARTITION_ID(DT_CHOSEN(zephyr_code_partition))
+#endif
+	) {
+		shell_error(sh, "Cannot erase boot or active partitions");
+		return 0;
+	}
```

Comment:
that's a bit rough imo :-) how about having two separate conditions?

```c
        /* Check if this is the parent (MCUboot) or own slot and if so, deny the request */
#if FIXED_PARTITION_EXISTS(boot_partition)
        if (id == FIXED_PARTITION_ID(boot_partition)) {
                shell_error(sh, "Cannot erase boot partition");
                return 0;
        }
#endif
#if DT_FIXED_PARTITION_EXISTS(DT_CHOSEN(zephyr_code_partition))
        if (id == DT_FIXED_PARTITION_ID(DT_CHOSEN(zephyr_code_partition))) {
                shell_error(sh, "Cannot erase active partitions");
                return 0;
        }
#endif
```

also is `return 0` correct?

Reply:
Have updated

Reply:
Thanks